asserts/snap_asserts.go

```diff
@@ -176,7 +183,53 @@ func snapDeclarationFormatAnalyze(headers map[string]interface{}, body []byte) (
 	return formatnum, nil
 }
 
-var validAlias = regexp.MustCompile("^[a-zA-Z0-9][-_.a-zA-Z0-9]*$")
+var (
+	validAlias   = regexp.MustCompile("^[a-zA-Z0-9][-_.a-zA-Z0-9]*$")
+	validAppName = regexp.MustCompile("^[a-zA-Z0-9](?:-?[a-zA-Z0-9])*$")
+)
+
+func checkAliases(headers map[string]interface{}) (map[string]string, error) {
+	value, ok := headers["aliases"]
+	if !ok {
+		return nil, nil
+	}
+	aliasList, ok := value.([]interface{})
+	if !ok {
+		return nil, fmt.Errorf(`"aliases" header must be a list of alias maps`)
+	}
+	if len(aliasList) == 0 {
+		return nil, nil
+	}
+
+	aliasMap := make(map[string]string, len(aliasList))
+	for i, item := range aliasList {
+		aliasItem, ok := item.(map[string]interface{})
+		if !ok {
+			return nil, fmt.Errorf(`"aliases" header must be a list of alias maps`)
+		}
+
+		what := fmt.Sprintf(`in "aliases" item %d`, i+1)
+		name, err := checkStringMatchesWhat(aliasItem, "name", what, validAlias)
+		if err != nil {
+			return nil, err
+		}
+
+		what = fmt.Sprintf(`for alias %q`, name)
+
```

Comment:
nitpick^100: maybe remove this \n as `what` and `target` seem to belong to each other (and its done the same in the previous block).